test/metabase/driver_test.clj

```diff
@@ -7,12 +7,26 @@
 (driver/register! ::test-driver, :abstract? true)
 
 (defmethod driver/supports? [::test-driver :foreign-keys] [_ _] true)
+(defmethod driver/database-supports? [::test-driver :foreign-keys] [_ _ db] (= db "dummy"))
+
 
 (deftest driver-supports?-test
   (is (= true
          (driver/supports? ::test-driver :foreign-keys)))
   (is (= false
-         (driver/supports? ::test-driver :expressions))))
+         (driver/supports? ::test-driver :expressions)))
+  (is (thrown? java.lang.Exception
+               (driver/supports? ::test-driver :some-made-up-thing))))
```

Comment:
Probably better to use `thrown-with-msg?` here so we can confirm that the exception thrown is actually the one we were expecting. For all we know something else is busted in the code and it's throwing an NPE; we wouldn't catch that with this test tho because it's only asserting that _some_ Exception is thrown